File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/non-plagiarized/09/t07.py
import sys

def sum_major_diagonal(matrix):
    result = 0
    for i in range(4):
        result += matrix[i][i]
    return result

if __name__ == "__main__":
    matrix = []
    print("Enter a 4-by-4 matrix row by row: ")
    for _ in range(4):
        row = [float(x) for x in input().split()]
        matrix.append(row)

    print(f"Sum of the elements in the major diagonal is {sum_major_diagonal(matrix)}")

File: datasets/IR-Plag/convertai-original-java-rest-kotlin/case-07/non-plagiarized/05/T07.kt
package IR

import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)
    val numbers = Array(4) { DoubleArray(4) }
    println("Enter a 4-by4 matrix row by row:")

    for (i in 0 until 4) {
        val rows = sc.nextLine()
        val rowsSplit = rows.split("\\s+".toRegex())
        for (j in 0 until 4) {
            numbers[i][j] = rowsSplit[j].toDouble()
        }
    }
    println("Sum of the elements in the major diagonal is ${sumMajorDiagonal(numbers)}")
}

fun sumMajorDiagonal(m: Array<DoubleArray>): Double {
    var jumDiagonal = 0.0
    for (i in 0 until 4) {
        for (j in 0 until 4) {
            if (i == j) {
                jumDiagonal += m[i][j]
            }
        }
    }
    return jumDiagonal
}

